src/celengine/universe.cpp

```diff
@@ -218,22 +196,22 @@ bool Universe::isMarked(const Selection& sel, int priority) const
 }
 
 
-class ClosestStarFinder : public StarHandler
+class ClosestStarFinder : public StarProcesor
 {
 public:
-    ClosestStarFinder(float _maxDistance, const Universe* _universe);
+    ClosestStarFinder(double _maxDistance, const Universe* _universe);
     ~ClosestStarFinder() = default;
-    void process(const Star& star, float distance, float appMag);
+    virtual void process(const Star *star, double distance, float appMag) override;
 
 public:
-    float maxDistance;
-    float closestDistance;
-    Star* closestStar;
+    double maxDistance;
```

Comment:
don't mix absolutely different things, if you want a double precision, make another PR

Reply:
It will be hard to revert. Moreover, it's already part of stars and dsos common interface for objects handled by octree. I see no reason to maintain separate versions of it, double and float....